File: platform/platform-core/src/main/java/ifml2/platform/core/fx/FilesTable.java
package ifml2.platform.core.fx;

import java.util.LinkedHashMap;

import org.osgi.framework.Bundle;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class FilesTable extends AbstractBundleEntryTable {

    Logger logger = LoggerFactory.getLogger(FilesTable.class);

    public FilesTable() {
        super();
    }

    @Override
    void bundleAdded(Bundle b) {
        // If the file was marked modified, mark it unchanged now:
//        BundleEntry be = getBundleEntry(b);
//        if (be!=null) {
//            be.setFileState(FileState.UNCHANGED);
//        }
    }

    @Override
    void bundleRemoved(Bundle bundle) {
        // Nothing to do here.
    }

    @Override
    void bundleModified(Bundle bundle) {
        BundleEntry be = getBundleEntry(bundle);
        if (be!=null) {
            be.fireBundleModified();
        }
    }

    public void updateWithWatchedFiles(LinkedHashMap<String, BundleEntry> changeSet) {
        logger.info("update...");
        for (BundleEntry wf : changeSet.values()) {
            BundleEntry be = null;
            switch (wf.getFileState()) {
            case ADDED:
                be = BundleEntry.createBundleEntry(wf.getFile());
                addBundleEntry(be);
                break;
            case REMOVED:
                be = getBundleEntry(wf.getFile());
                if (be!=null) {
                    removeBundleEntry(be);
                }
                break;
            case MODIFIED:
                be = getBundleEntry(wf.getFile());
                if (be!=null) {
                    be.setFileState(BundleEntry.FileState.MODIFIED);
                }
                break;
            case CURRENT:
                break;
            default:
            }
        }
        logger.info("update... done.");
    }

}

File: platform/platform-core/src/main/kotlin/ifml2/platform/core/fx/FilesTable.kt
package ifml2.platform.core.fx

import org.osgi.framework.Bundle
import org.slf4j.LoggerFactory
import java.util.*

class FilesTable : AbstractBundleEntryTable() {

    internal override var logger = LoggerFactory.getLogger(FilesTable::class.java)

    internal override fun bundleAdded(bundle: Bundle) {
        // If the file was marked modified, mark it unchanged now:
        //        BundleEntry be = getBundleEntry(b);
        //        if (be!=null) {
        //            be.setFileState(FileState.UNCHANGED);
        //        }
    }

    internal override fun bundleRemoved(bundle: Bundle) {
        // Nothing to do here.
    }

    internal override fun bundleModified(bundle: Bundle) {
        val be = getBundleEntry(bundle)
        be?.fireBundleModified()
    }

    fun updateWithWatchedFiles(changeSet: LinkedHashMap<String, BundleEntry>) {
        logger.info("update...")
        for (wf in changeSet.values) {
            var be: BundleEntry? = null
            when (wf.getFileState()) {
                BundleEntry.FileState.ADDED -> {
                    be = BundleEntry.createBundleEntry(wf.file.value)
                    addBundleEntry(be)
                }
                BundleEntry.FileState.REMOVED -> {
                    be = getBundleEntry(wf.file.value)
                    if (be != null) {
                        removeBundleEntry(be)
                    }
                }
                BundleEntry.FileState.MODIFIED -> {
                    be = getBundleEntry(wf.file.value)
                    if (be != null) {
                        be.setFileState(BundleEntry.FileState.MODIFIED)
                    }
                }
                BundleEntry.FileState.CURRENT -> {
                }
            }
        }
        logger.info("update... done.")
    }

}
